presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -620,41 +617,63 @@ private static String keyFromPath(Path path)
         return key;
     }
 
-    private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig, ClientConfiguration clientConfig)
+    private AmazonS3 createAmazonS3Client(URI uri, Configuration hadoopConfig, ClientConfiguration clientConfig)
     {
         AWSCredentialsProvider credentials = getAwsCredentialsProvider(uri, hadoopConfig);
-        Optional<EncryptionMaterialsProvider> emp = createEncryptionMaterialsProvider(hadoopConfig);
-        AmazonS3Client client;
+        Optional<EncryptionMaterialsProvider> encryptionMaterialsProvider = createEncryptionMaterialsProvider(hadoopConfig);
+        AmazonS3Builder<? extends AmazonS3Builder, ? extends AmazonS3> clientBuilder;
+
         String signerType = hadoopConfig.get(S3_SIGNER_TYPE);
         if (signerType != null) {
             clientConfig.withSignerOverride(signerType);
         }
-        if (emp.isPresent()) {
-            client = new AmazonS3EncryptionClient(credentials, emp.get(), clientConfig, new CryptoConfiguration(), METRIC_COLLECTOR);
+
+        if (encryptionMaterialsProvider.isPresent()) {
+            clientBuilder = AmazonS3EncryptionClient.encryptionBuilder()
+                    .withCredentials(credentials)
+                    .withEncryptionMaterials(encryptionMaterialsProvider.get())
+                    .withClientConfiguration(clientConfig)
+                    .withMetricsCollector(METRIC_COLLECTOR);
         }
         else {
-            client = new AmazonS3Client(credentials, clientConfig, METRIC_COLLECTOR);
+            clientBuilder = AmazonS3Client.builder()
+                    .withCredentials(credentials)
+                    .withClientConfiguration(clientConfig)
+                    .withMetricsCollector(METRIC_COLLECTOR);
         }
 
-        if (isPathStyleAccess) {
-            S3ClientOptions clientOptions = S3ClientOptions.builder().setPathStyleAccess(true).build();
-            client.setS3ClientOptions(clientOptions);
-        }
+        boolean regionOrEndpointSet = false;
 
         // use local region when running inside of EC2
         if (pinS3ClientToCurrentRegion) {
             Region region = Regions.getCurrentRegion();
             if (region != null) {
-                client.setRegion(region);
+                clientBuilder = clientBuilder.withRegion(region.getName());
+                regionOrEndpointSet = true;
             }
         }
 
         String endpoint = hadoopConfig.get(S3_ENDPOINT);
         if (endpoint != null) {
-            client.setEndpoint(endpoint);
+            if (regionOrEndpointSet) {
+                log.warn("Ignoring the S3 endpoint configuration as region has already been set by pinning the S3 client to the current region.");
```

Comment:
What about considering this an invalid configuration and throwing instead?

Reply:
Yep, it makes sense to throw in this case. I added a `verify()` to `initialize()` for now.